content/en/blog/_posts/2022-12-16-non-graceful-node-shutdown-to-beta.md

```diff
@@ -0,0 +1,80 @@
+
+## What’s new in Beta
+
+With the promotion of the Non-Graceful Node Shutdown feature to beta, the feature gate `NodeOutOfServiceVolumeDetach` is enabled by default on `kube-controller-manager` instead of being opt-in.
+
+New metrics `force_delete_pods_total`, number of pods that are being forcefully deleted since the Pod GC Controller started, and `force_delete_pod_errors_total`, number of errors encountered when forcefully deleting the pods since the Pod GC Controller started, are also added in the Pod GC Controller.
+
+## How does it work
+
+In the case of a node shutdown, if a graceful shutdown is not working or the node is in a non-recoverable state due to hardware failure or broken OS, the user can manually add an `out-of-service` taint on the Node. For example, this can be `node.kubernetes.io/out-of-service=nodeshutdown:NoExecute` or `node.kubernetes.io/out-of-service=nodeshutdown:NoSchedule`. This will trigger pods on the node to be forcefully deleted if there are no matching tolerations on the pods.  Persistent volumes attached to the shutdown node will be detached, and new pods will be created successfully on a different running node.
+
+```
+kubectl taint nodes <node-name> node.kubernetes.io/out-of-service=nodeshutdown:NoExecute
+```
+
+Note: Before applying the out-of-service taint, you must verify that a node is already in shutdown or power-off state (not in the middle of restarting), either because the user intentionally shut it down or the node is down due to hardware failures, OS issues, etc.
+
+Once all the workload pods that are linked to the out-of-service node are moved to a new running node, and the shutdown node has been recovered, you should remove that taint on the affected node after the node is recovered.
+
+## What’s next?
+
+Depending on feedback and adoption, the Kubernetes team plans to push the Non-Graceful Node Shutdown implementation to GA in either 1.27 or 1.28.
+
+This feature requires a user to manually add a taint to the node to trigger the failover of workloads and remove the taint after the node is recovered. In the future, we plan to find ways to automatically detect and fence nodes that are shut down or in a non-recoverable state and fail their workloads over to another node.
```

Comment:
Can we say anything about a cluster operator using their own automation to add that taint? Sometimes there is an API that lets you confirm that a node is terminated.
Any guidelines? (ideally those would also be in the docs)

Reply:
There are already solutions from storage vendors that validate if host is still connected to storage and if there is IO.  If not, fence and cleanup from the storage side without a reboot. There are also solutions that monitor the node status and require the nodes to be rebooted. There are also people who want their applications to be opted out of this process.  We have not reached consensus on a common approach yet.

If you think it is helpful, I can add a line saying that "the cluster operator can automate this process if there is a programmatic way to determine that the node is really shut down and there isn’t IO between the node and storage".